shared/lib/tasks.py

```diff
@@ -19,7 +19,22 @@ def __init__(self, fn, args=None, kwargs=None, timeout=0, repeated=False, immedi
 		self.fn = fn
 		self.args = args
 		self.kwargs = kwargs
-		self.next = time.time() + (0 if immediate else timeout) 
+
+		# if random offset to be used as first timeout, next_timeout must be randomly selected.
+		if random_offset:
+			next_time = random.randint(0, timeout)
+		else:
+			next_time = timeout
+
+		# if there is an additional, immediate execution, next_time has to be used as next_timeout, and next_time must be 0.
+		if immediate:
+			self.next_timeout = next_time  # will be overwritten by self.timeout after first execute()
```

Comment:
```next_time``` determines the first execution.
```self.next_timeout``` determines the time between first and second execution.
```self.timeout``` determines the time between all other execution (2nd-3rd, 3rd-4th, ...)

next_timeout is needed. 4 cases:

* not immediate and not random_offset:
 * first execution at now+timeout, then repeat every timeout.
 * next_time == timeout and self.next_timeout == timeout
* not immediate and random_offset:
 * first execution at now + random_time, then repeat every timeout.
 * next_time == random() and self.next_timeout == timeout
* immediate and not random_offset:
 * first execution immediately. second execution a timeout later. then repeat every timeout.
 * next_time == 0 and self.next_timeout == timeout
* immediate and random_offset:
 * first execution immediately. second execution a random_time later. then repeat every timeout.
 * next_time == 0 and self.next_timeout == random()

self.next_timeout will be set to self.timeout in line 52. So this is used exactly once when rescheduling the Task after its execution in line 51.